parachains/common/src/process_xcm_message.rs

```diff
@@ -0,0 +1,118 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Implementation of `ProcessMessage` for an `ExecuteXcm` implementation.
+
+use codec::{Decode, FullCodec, MaxEncodedLen};
+use cumulus_primitives_core::{AggregateMessageOrigin, ParaId};
+use frame_support::{
```

Comment:
Why is this called `AggregateMessageOrigin`, namely what is `Aggregate` supposed to mean?

Reply:
probably because of https://github.com/paritytech/polkadot/blob/b749009549428344e6aec63d4a581a6a6e688af6/runtime/parachains/src/inclusion/mod.rs#L240?

Reply:
The idea is the keep it extensible since we soon also want `Loopback` and `Bridged(…)` origins. So it aggregates all of them.